### PR TITLE
add the ability to define default launcher fields

### LINKED ISSUE #3099: script configurable required items
In the original comment below we're talking about what fields are required for submitting a job. At a minimum `script` and `cluster` must be populated to submit a job to any scheduler at any center.

However, at OSC, we'd also require an `account`. (unless you have a slurm default file which shouldn't be a prerequisite for most users)

So, let's add a configuration that specifies what fields cannot be removed in `script#edit` so that centers (like OSC) are always populating what they need when they make new scripts.

_Originally posted by @johrstrom in https://github.com/OSC/ondemand/issues/3095#issuecomment-1746837806_

### PR DESCRIPTION
**Pull request by johrstrom**
Fixes #3099 by allowing centers to define launcher fields that should get created when you create a new launcher.

### CODE REVIEW

File: apps/dashboard/app/models/launcher.rb [hunk 1]
```diff
@@ -353,6 +355,13 @@ def add_required_fields(form: [], attributes: {})
     add_script_to_form(form: form, attributes: attributes)
   end
 
+  def add_default_fields(form: [], **_args)
+    Configuration.launcher_default_items.each do |default_item|
+      Rails.logger.debug("adding #{default_item} unless #{form.include?(default_item)}")
```
Comment (Oglopf) [inline apps/dashboard/app/models/launcher.rb:360]: Did you want to leave this `debug` in or was it supposed to be yanked?

**Review by Oglopf — DISMISSED** [review]
This looks good, left a comment about `debug` but works for me.

File: apps/dashboard/app/models/launcher.rb (hunk 1 above)
Comment (johrstrom) [reply apps/dashboard/app/models/launcher.rb:360]: Good catch, I'll remove it.

**Review by Oglopf — APPROVED** [review]
Looks good!

**Pull request merged**